src/core_plugins/kibana/public/management/sections/indices/create_index_pattern_wizard/step_index_pattern/append_wildcard.js

```diff
@@ -0,0 +1,53 @@
+import { uiModules } from 'ui/modules';
+
+const module = uiModules.get('apps/management');
+
+/**
+ * This directive automatically appends a wildcard to the input field
+ * after the user starts typing. It lets the user delete the wildcard
+ * if necessary. If the value of the input field is set back to an empty
+ * string, the wildcard is immediately re-appended after the user starts
+ * typing. This is intended to be a UX improvement for the index pattern
+ * creation page. See https://github.com/elastic/kibana/pull/13454
+ */
+module.directive('appendWildcard', function () {
+  return {
+    require: 'ngModel',
+    link: function ($scope, $elem, $attrs, $ctrl) {
+      $scope.appendedWildcard = false;
+
+      function setElementValue(val, event, moveSelectionRange = false) {
+        event.preventDefault();
+        $elem.val(val);
+        if (moveSelectionRange) {
+          $elem[0].setSelectionRange(1, 1);
```

Comment:
I had to think for a minute to grok why this line is doing what it does.  Maybe parameter could be named better to make that clear.

Reply:
On it!